packages/connectkit/src/components/DaimoPayButton/index.tsx

```diff
@@ -1,128 +1,217 @@
 import React, { useEffect } from "react";
-import { useAccount, useEnsName } from "wagmi";
 import useIsMounted from "../../hooks/useIsMounted";
-import { truncateEthAddress } from "./../../utils";
 
 import { usePayContext } from "../DaimoPay";
 import { TextContainer } from "./styles";
 
 import { AnimatePresence, Variants } from "framer-motion";
-import { Chain } from "viem";
-import { useChainIsSupported } from "../../hooks/useChainIsSupported";
+import { Address, Hex } from "viem";
 import { ResetContainer } from "../../styles";
-import { CustomTheme, Mode, Theme } from "../../types";
+import { CustomTheme, Mode, PaymentOption, Theme } from "../../types";
 import ThemedButton, { ThemeContainer } from "../Common/ThemedButton";
 
-const contentVariants: Variants = {
-  initial: {
-    zIndex: 2,
-    opacity: 0,
-    x: "-100%",
-  },
-  animate: {
-    opacity: 1,
-    x: 0.1,
-    transition: {
-      duration: 0.4,
-      ease: [0.25, 1, 0.5, 1],
-    },
-  },
-  exit: {
-    zIndex: 1,
-    opacity: 0,
-    x: "-100%",
-    pointerEvents: "none",
-    position: "absolute",
-    transition: {
-      duration: 0.4,
-      ease: [0.25, 1, 0.5, 1],
-    },
-  },
-};
+type PayButtonCommonProps =
+  | {
+      /**
+       * Your public app ID. Specify either (payId) or (appId + parameters).
+       */
+      appId: string;
+      /**
+       * Optional nonce. If set, generates a deterministic payID. See docs.
+       */
+      nonce?: bigint;
```

Comment:
this is the nonce as in `destNonce` onchain? Currently we always set nonce = payID onchain, and it's a nice invariant to enforce that the create2 parameters are publicly known by all parties. I'd recommend keeping that / not providing this option perhaps?

Reply:
invariant is still there. if you provide the nonce, that becomes the id. (Wrote "generates a determinstic payID" to avoid confusion because our payIds are formatted in base58.)

Reply:
maybe there's a cleaner way to present that.

example use case in DonateForm:
- generate nonce
- on click, save (form info, payment nonce)
- then show checkout modal

this creates a guarantee that every payment is attributed. useful for eg cart checkout too.

alternative way:

`newPayId` = ... and we provide a utility function `createRandomPayID` that returns the base58 form

then we don't have to expose concept of "nonce" in SDK

lmk if that feels cleaner

Reply:
update to use `newPayId` to reduce confusion

Reply:
deleted `newPayId`, just using onPaymentStarted for now